Source/WebInspectorUI/UserInterface/Controllers/DebuggerManager.js

```diff
@@ -131,9 +133,15 @@ WI.DebuggerManager = class DebuggerManager extends WI.Object
                 const key = null;
                 WI.objectStores.breakpoints.associateObject(breakpoint, key, serializedBreakpoint);
 
+                let sourceCode = this._sourceCodeContentIdentifierMap.get(breakpoint.contentIdentifier);
+                if (sourceCode)
+                    this._associateBreakpointsWithSourceCode([breakpoint], sourceCode);
+
                 this.addBreakpoint(breakpoint);
             }
             this._restoringBreakpoints = false;
+            this._hasLoadedExistingJavaScriptBreakpoints = true;
+            this._sourceCodeContentIdentifierMap.clear();
```

Comment:
NIT: i wonder if we should `= null` instead since we're never going to use it again

we could even use that instead of `_hasLoadedExistingJavaScriptBreakpoints` in `breakpointsForSourceCode`
```js
this._sourceCodeContentIdentifierMap?.set(sourceCode.contentIdentifier, sourceCode);
```